src/output.rs

```diff
@@ -238,6 +275,46 @@ impl Output {
         Self::new(ContentPredicate::Is(pred))
     }
 
+    /// Expect the command to **match** this `output`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// assert_cli::Assert::command(&["echo"])
+    ///     .with_args(&["42"])
+    ///     .stdout().matches("[0-9]{2}")
+    ///     .unwrap();
+    /// ```
+    pub fn matches(output: String) -> Self {
+        let pred = RegexPredicate {
+            regex: regex::Regex::new(&output).unwrap(),
+            times: 0,
+        };
+        Self::new(ContentPredicate::Regex(pred))
+    }
+
+    /// Expect the command to **match** this `output` exacly `nmatches` times.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// assert_cli::Assert::command(&["echo"])
+    ///     .with_args(&["42"])
+    ///     .stdout().matches_ntimes("[0-9]{1}", 2)
+    ///     .unwrap();
+    /// ```
+    pub fn matches_ntimes(output: String, nmatches: u32) -> Self {
```

Comment:
(longer term brain storming, feel free to ignore me)

Ideally, in the long run I'd like to do this through a more builder-like approach `.matches("pattern").times(10)`.  `contains` could also implement this.  It'd provide a nice way to keep the upfront API "small".  We're already talking about doing this for other [features](https://github.com/killercup/assert_cli/issues/77#issuecomment-341932465).

The interesting challenge is deciding how to implement that.

The nasty "unsafe" option is for `Output` to have a `.times()` function.  I say "unsafe" because the call is meaningless in some cases (like `.is`).

Another option is to move away from having people interact with `Output` and instead have people construct the predicates directly and we'll do an `Into<ContentPredicate>`.  This will end up more like killercup's proposed API.

Reply:
This is something that can be very useful but out of the scope of this PR, for now, `match_ntimes(V, N)` is a good enough solution, we might as well open an improvement for later to refactor and abstract into `times(N)`.